### PR TITLE
End points

### PR DESCRIPTION
**Pull request by mrparkersson**
-Create an API endpoint to list all posts for a user.
-Create an API endpoint to list all comments for a user's post.
-Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
-Add authentication to your API endpoints.
-Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
 
   before_action :update_allowed_parameters, if: :devise_controller?
+  before_action :authenticate_request
+
+  def authenticate_request
+    header = request.headers['Authorization']
+    header = header.split.last if header
+    decoded = jwt_decode(header)
+    @curr_user = User.find(decoded[:user_id])
+  end
```
Comment (DeliceLydia) [inline app/controllers/application_controller.rb:13]: - [ ] I am not able to sign in it seems like there is an error while decoding a token please address that
token

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Highlights for your project:
- Awesome README
- Followed git workflow


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (DeliceLydia) [inline app/controllers/application_controller.rb:13]: - [ ] I am not able to sign up or sign in there is an error while decoding a token please address that
token

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @mrparkersson, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️ Great job so far

 ## Required Changes ♻️
- [ ] I couldn't run your app as it throws an error in the terminal when I run rails db:create; 
As I couldn't run I was not able to check previous reviewers' requirements to sign up or sign in.
Screenshot 2022-06-18 at 3 24 37 PM

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mrparkersson** [comment]
Now  rails db:create work
Screen Shot 2022-06-18 at 1 22 34 PM
s

**Review by dasileker — APPROVED** [review]
### **Hi @mrparkersson,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**